src/Microsoft.VisualStudio.Services.Agent/Util/ApiUtil.cs

```diff
@@ -63,6 +64,7 @@ public static VssCredentials GetVssCredential(ServiceEndpoint serviceEndpoint)
             ArgUtil.NotNull(serviceEndpoint.Authorization, nameof(serviceEndpoint.Authorization));
             ArgUtil.NotNullOrEmpty(serviceEndpoint.Authorization.Scheme, nameof(serviceEndpoint.Authorization.Scheme));
 
+            WebProxy.ApplyProxySettings();
```

Comment:
will move the call to ApplyProxySettings in Agent.ExecuteCommand and in Worker.RunAsync, which will also make sure there are no race conditions.